gradient_boosting.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import RegressorMixin
+from tree import DecisionTreeRegressor
+
+
+class BaseGradientBoostingRegressor:
+
+    def __init__(self,
+                 n_estimators: 5,
+                 max_depth: int,
```

Comment:
Please, set `None` as a default parameter for `max_depth`, which means no constraints on depth.